mtags/src/main/scala-2/scala/meta/internal/mtags/MtagsEnrichments.scala

```diff
@@ -129,28 +130,44 @@ trait MtagsEnrichments extends CommonMtagsEnrichments {
     def stripBackticks: String = value.stripPrefix("`").stripSuffix("`")
     def isBackticked: Boolean =
       value.size > 1 && value.head == '`' && value.last == '`'
-    def toAbsolutePath: AbsolutePath = toAbsolutePath(true)
+    def toAbsolutePath: AbsolutePath = {
+      toAbsolutePath(true)
+    }
     def toAbsolutePath(followSymlink: Boolean): AbsolutePath = {
-      // Windows sometimes treats % literally:
-      // https://learn.microsoft.com/en-us/troubleshoot/windows-client/networking/url-encoding-unc-paths-not-url-decoded
-      def decoded =
-        if (Properties.isWin) URIEncoderDecoder.decode(value) else value
 
-      // jar schemes must have "jar:file:"" instead of "jar:file%3A" or jar file system won't recognise the URI.
-      // but don't overdecode as URIs may not be recognised e.g. "com-microsoft-java-debug-core-0.32.0%2B1.jar" is correct
-      if (value.toUpperCase.startsWith("JAR%3AFILE"))
-        URLDecoder.decode(decoded, "UTF-8").toAbsolutePath(followSymlink)
-      else if (value.toUpperCase.startsWith("JAR:FILE%3A"))
-        URLDecoder.decode(decoded, "UTF-8").toAbsolutePath(followSymlink)
-      else if (value.toUpperCase.startsWith("JAR")) {
+      /* Windows sometimes treats % literally, but not sure if it's always the case.
+       * https://learn.microsoft.com/en-us/troubleshoot/windows-client/networking/url-encoding-unc-paths-not-url-decoded
+       * This function tries to apply different heuristics to get the proper file system.
+       */
+      def withTryDecode(value: String)(f: String => AbsolutePath) = {
         try {
-          new URI("jar", decoded.stripPrefix("jar:"), null)
-            .toAbsolutePath(followSymlink)
+          if (Properties.isWin) f(URLDecoder.decode(value, "UTF-8"))
```

Comment:
Freaking windows.

Reply:
I am sort of trying all the possibilities because it seems every time one thing is fixed the other breaks, so instead let's hope fallback work.